packages/flutter/lib/src/foundation/observer_list.dart

```diff
@@ -29,7 +27,7 @@ import 'dart:collection';
 class ObserverList<T> extends Iterable<T> {
   final List<T> _list = <T>[];
   bool _isDirty = false;
-  HashSet<T> _set;
+  late final HashSet<T> _set = HashSet<T>();
```

Comment:
Why `late`?

Reply:
IIRC this will make the field lazy initialized.

Reply:
It won't do anything in this case. It allows you to (yourself) lazily initialize it but leave it as a non-nullable type.

Reply:
Sorry if this wasn't clear, but my question was meant as: "Why are we declaring this field as lazy? It seems unnecessary here."

Reply:
It was lazy before - I think there was a misunderstanding around what `late` means. If assigning immediately it has no meaning @stereotype441 should we have a lint for this?

To restore the lazy behavior that existed before this would have to be a nullable field type, and you would have to restore the old logic.

Reply:
@jakemac53 that's not correct. late with an initializer means lazy.

Reply:
Now I am confused. What is the difference between those two lines?

```dart
late final int foo = 1;
final int bar = 1;
```

Reply:
Nothing, because you can't observe laziness with a value.  But:

```dart
class A {
  late final int foo = (() {print("hello"); return 1;})();
  final int bar = (() {print("world"); return 1;})();
  A();
}
                            
                            
void main() {
  var a = A();
  print("says");
  a.foo;
}
```

outputs
```
world
says
hello
```

because a `late` variable with an initializer only evaluates its initializer on first use.

cc @munificent this is something we're going to have to explain carefully.

Reply:
Thanks for the explanation!